webanno-security/src/main/java/de/tudarmstadt/ukp/clarin/webanno/security/session/SessionListener.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2019
+ * Ubiquitous Knowledge Processing (UKP) Lab
+ * Technische Universität Darmstadt
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package de.tudarmstadt.ukp.clarin.webanno.security.session;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.ApplicationEvent;
+import org.springframework.context.event.EventListener;
+import org.springframework.core.Ordered;
+import org.springframework.core.annotation.Order;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.session.SessionRegistry;
+import org.springframework.security.web.session.HttpSessionCreatedEvent;
+import org.springframework.security.web.session.HttpSessionDestroyedEvent;
+import org.springframework.stereotype.Component;
+
+/**
+ * Listens to HTTP sessions being created and destroyed and (un)registers accordingly in the
+ * {@link SessionRegistry}. This is mainly required when using pre-authentication since the
+ * login page usually takes care of registering the session.
+ */
+@Component
+public class SessionListener
+{
+    private final Logger log = LoggerFactory.getLogger(getClass());
+    
+    private final SessionRegistry sessionRegistry;
+
+    public SessionListener(SessionRegistry aSessionRegistry)
```

Comment:
Why doesnt this need `@autowired`?

Reply:
Good question. It seems to work. Probably because there is only one constructor.